lib/publiccloud/provider.pm

```diff
@@ -230,37 +220,87 @@ sub create_instance {
     $args{instance_type} //= get_var('PUBLIC_CLOUD_INSTANCE_TYPE');
     $args{image} //= $self->get_image_id();
 
+    my @vms = $self->terraform_apply(count => '1');
+    my $instance = $vms[0];
+    record_info('INFO', "VM created '" . Dumper $instance);
+
+    my $timeout = 60;
+    while ($timeout > 0) {
+        last if script_run("nc -vz -w 1 $instance->{public_ip} 22") == 0;
+        $timeout -= 1;
+        sleep 5;
+    }
+    die('Cannot reach port 22 on IP ' . $instance->{public_ip}) if ($timeout == 0);
+    return $instance;
+}
+
+=head2 terraform_apply
+
+Calls terraform tool and applies the corresponding configuration .tf file
+
+=cut
+sub terraform_apply {
+    my ($self, %args) = @_;
+    my @instances;
+
```

Comment:
Should we add a sanity check, so calling this function twice is forbidden, without calling terraform_destroy() in between?

Reply:
I added a warning. Calling this function twice is not desired but terraform command will return 0 anyways, indicating that the resources are already created.